database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractAnnotationFromVariant.java

```diff
@@ -0,0 +1,346 @@
+                            + "). The '" + ANNOT_FIELD + "' field will not be removed from the "
+                            + variantsCollection.getNamespace() + " collection.");
+        }
+    }
+
+    /**
+     * Return a batch of elements, advancing the Iterator provided.
+     * @param iterator won't be closed, please close it outside this function.
+     * @param bulkSize maximum size for the batch. The list returned can be smaller.
+     * @return A list with elements, or an empty list if there are no more elements in the iterator.
+     */
+    private <T> List<T> getBatch(Iterator<T> iterator, int bulkSize) {
+        List<T> batch = new ArrayList<>();
+        int counter = 0;
+        while (iterator.hasNext()) {
+            T element = iterator.next();
+            if (element != null) {
+                counter++;
+                batch.add(element);
+                if (counter % bulkSize == 0) {
+                    return batch;
+                }
+            }
+        }
+        return batch;
+    }
+
+    private InsertOneModel<Document> getInsertionDocument(Document variantDocument) {
+        Document annotationSubdocument = (Document) variantDocument.get(ANNOT_FIELD);
+        if (annotationSubdocument == null) {
+            return null;
+        } else {
+            return buildInsertionDocument(variantDocument, annotationSubdocument);
+        }
+    }
+
+    private InsertOneModel<Document> buildInsertionDocument(Document variantDocument, Document annotationSubdocument) {
+        annotationSubdocument.put(ID_FIELD, buildAnnotationId(variantDocument));
+        annotationSubdocument.put(CHROMOSOME_FIELD, variantDocument.get("chr"));
+        annotationSubdocument.put(START_FIELD, variantDocument.get("start"));
+        annotationSubdocument.put(END_FIELD, variantDocument.get("end"));
+        annotationSubdocument.put(VEP_VERSION_FIELD, databaseParameters.getVepVersion());
+        annotationSubdocument.put(CACHE_VERSION_FIELD, databaseParameters.getVepCacheVersion());
+        return new InsertOneModel<>(annotationSubdocument);
+    }
+
+    private String buildAnnotationId(Document variantDocument) {
+        return variantDocument.get("_id")
+                + "_" + databaseParameters.getVepVersion()
+                + "_" + databaseParameters.getVepCacheVersion();
+    }
+
+    @ChangeSet(order = "002", id = "reduceAnnotationFromVariants", author = "EVA")
+    public void reduceAnnotationFromVariants(MongoDatabase mongoDatabase) {
+        final MongoCollection<Document> variantsCollection = mongoDatabase.getCollection(
+                databaseParameters.getDbCollectionsVariantsName());
+        logger.info("2) reduce annotation field from collection {}", variantsCollection.getNamespace());
+
+        variantsCollection.dropIndex(VARIANTS_XREFS_INDEX);
+        variantsCollection.dropIndex(VARIANTS_SO_INDEX);
```

Comment:
The indexes are originally in `annot.xrefs.id` and `annot.ct.so`.

The new ones will be `annot.xrefs` and `annot.so` because of the field being flatten.